### PR TITLE
Increase Test Coverage, edit User Guide

### LINKED ISSUE #248: Student can be put in Participants and Blacklist of an event at the same time through manual edits

### PR DESCRIPTION
**Pull request by TLChicken**
Fixes and resolves #248.

### CODE REVIEW

File: src/test/java/nustracker/logic/commands/EnrollCommandTest.java [hunk 1]
```diff
@@ -1,121 +1,267 @@
...
+
+            CommandException expectedCommandException = new CommandException(String.format(
+                    MESSAGE_STUDENT_ALREADY_ENROLLED,
+                    HANS.getName().toString(),
+                    HANS.getStudentId().getStudentIdString(),
+                    MATH_OLYMPIAD.getName().getEventName()));
+
+            assertEquals(expectedCommandException.toString(), actualCommandException.toString());
+            return;
+        }
```
Comment (Joel-Sung) [inline src/test/java/nustracker/logic/commands/EnrollCommandTest.java:111]: You can use assertThrows to test for exceptions. It takes in the class of exception, the error string and the executable as 3 parameters.

File: src/test/java/nustracker/logic/commands/EnrollCommandTest.java [hunk 2]
```diff
@@ -1,121 +1,267 @@
...
+
+        compareModel.addEnrolledEvent(currStudent, currEvent);
+        compareModel.addParticipant(currEvent, STUDENTID_ONE);
+
+        CommandResult actualCommandResult = null;
+
+        try {
+            actualCommandResult = enrollCommand.execute(model, CurrentlyShownList.STUDENTS_LIST);
+        } catch (CommandException e) {
+            fail("Command Exception should not be thrown.");
```
Comment (Joel-Sung) [inline src/test/java/nustracker/logic/commands/EnrollCommandTest.java:80]: I'm not sure what this fail method is, but we have not used it throughout the other test cases. Maybe if not necessary can avoid using it? Unless it works better then go ahead.

File: src/test/java/nustracker/testutil/TypicalEvents.java [hunk 3]
```diff
@@ -30,7 +30,7 @@
             .withDate("10-09-2021")
             .withTime("1200")
             .withParticipants("e9831818")
-            .withBlacklist("e0456789").build();
+            .withBlacklist("e0456789", "e0123456").build();
```
Comment (Joel-Sung) [inline src/test/java/nustracker/testutil/TypicalEvents.java:33]: Ensure this is consistent with the other JSON files.

**Review by Joel-Sung — COMMENTED** [review]
Consider using assertThrows for your test cases.

File: src/test/java/nustracker/logic/commands/EnrollCommandTest.java (hunk 1 above)
Comment (TLChicken) [reply src/test/java/nustracker/logic/commands/EnrollCommandTest.java:111]: Ok I'll try that

File: src/test/java/nustracker/logic/commands/EnrollCommandTest.java (hunk 2 above)
Comment (TLChicken) [reply src/test/java/nustracker/logic/commands/EnrollCommandTest.java:80]: The fail(String message) method just triggers a fail for the current test with a message when it runs and is one of the assertion methods in JUnit just like assertEquals, so it should be ok to use: https://junit.org/junit5/docs/5.4.0-RC2/api/org/junit/jupiter/api/Assertions.html

**Review by Joel-Sung — APPROVED** [review]
LGTM.

**Pull request merged**